semantic-core/src/Analysis/Analysis.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE RankNTypes #-}
+module Analysis.Analysis
+( Analysis(..)
+) where
+
+import Core.Name
+import Data.Text (Text)
+
+-- | A record of functions necessary to perform analysis.
+--
+-- This is intended to be replaced with a selection of algebraic effects providing these interfaces and carriers providing reusable implementations.
```

Comment:
As mentioned here, `Analysis` has only ever been defined for expedience’s sake; I intend to split it into effects and carriers in a future PR.